src/main/java/hudson/plugins/ec2/SpotConfiguration.java

```diff
@@ -20,6 +20,19 @@
         }
     }
 
+    /**
+     * Export the spotBlockReservationDuration attribute for CasC plugin.
+     *
+     * @return The spotBlockReservationDuration attribute as a string.
+     */
+    public String getSpotBlockReservationDurationStr() {
+        if (spotBlockReservationDuration == Integer.MAX_VALUE) {
```

Comment:
why is this if statement required?

while this would probably work a couple of suggestions but feel free to ignore if maintainers just want the quick fix...

1. The current constructor isn't a 'correct' `DataBoundConstructor`, it has 'mandatory' fields that aren't used in databinding, i.e. `spotBlockReservationDurationStr` is bound to `spotBlockReservationDuration`

This could be solved by deprecating the existing constructor, fixing the parameters in a new constructor and updating any calling code / views.

2. It would be good to extend the test to verify that this works and it doesn't get broken in the future, here's the existing test: https://github.com/jenkinsci/ec2-plugin/blob/master/src/test/java/hudson/plugins/ec2/ConfigurationAsCodeTest.java

Reply:
I kinda agree with @timja fix the databound constructor.

Make if follow java beans standards not even sure why it takes in strings when stapler and jelly have full support for numbers.

Reply:
btw while your at it make the data bound constructor only have mandatory fields and use data bound setters for optional fields.
See https://issues.jenkins-ci.org/browse/JENKINS-57513